doc/design/katib_on_sqlflow.md

```diff
@@ -64,28 +65,59 @@ We need to develop a new codegen, `codegen_couler.go`, for SQLFlow.  `codegen_co
 
 ### The Integration via Couler
 
-SQLFlow parses each SQL program into an IR, which is a list of statement IRs.  The `codegen_couler.go` converts the IR into a Couler program.   We need to add a Couler functions `couler.katib.train` for the calling by the generated Couler program.
+SQLFlow parses each SQL program into an IR, which is a list of statement IRs.  The `codegen_couler.go` converts the IR into a Couler program.   We need to add a Couler functions `couler.sqlflow.train` for the calling by the generated Couler program.
 
 Consider the following example program.
 
 ```sql
 SELECT * FROM a, b WHERE a.id = b.id INTO c;
-SELECT * FROM c TO TRAIN model_def WITH objective=multi:softmax, eta=1 LABEL class INTO my_xgb_model;
+SELECT * FROM c TO TRAIN data_scientist/xgboost:v0.5/xgboost.gbtree 
+    WITH objective=multi:softmax, eta=0.1, validation_dataset="select * from d;" 
+    INTO my_xgb_model;
 ```
 
 The `codegen_couler.go` might generate the following Couler program.
 
 ```python
-couler.maxcompute.run("""SELECT * FROM a, b WHERE a.id = b.id INTO c;
-                         SELECT * FROM c INTO temp""")
-couler.maxcompute.export(table="temp", file="/hdfs/temp")
-couler.katib.train(model_def, data="/hdfs/temp")
+couler.maxcompute.run("""SELECT * FROM a, b WHERE a.id = b.id INTO c;""")
+couler.sqlflow.train(model="xgboosy", hyperparameters={ "booster": "gbtree", "objective": "multi:softmax", 
+    "eta": 0.1},  image="data_scientist/xgboost:v0.5",
+    sql="select * from c to train data_scientist/xgboost:v0.5/xgboost.gbtree ... ",
+    datasource="mysql:// ...")
 ```
 
-## `couler.katib.train(...)`
+## `couler.sqlflow.train(...)`
 
 Considering Katib itself supports multiple models and frameworks, and more may come in the future, we introduce the following Couler function.
 
 ```python
-def couler.katib.train(model_def=None, hyperparameters={})
+def couler.sqlflow.train(model=None, hyperparameters={}, image=None, sql=None,datasource=None)
 ```
+
+The arguments in `couler.sqlflow.train`,
+
+- `model` defines the training model, e.g., `xgboost`.
+- `hyperparameters` specifies hyperparameters for model given in `model`.
+- `image` specifies the container image source for the Katib tuning job.
+- `sql` sql statement input by users.
+- `datasource` train and validation data source 
+
+## Run Tuning Job on Katib
+
+In each Katib tuning job, users need to define tuning parameters (i.e., the hyperparameter's name, type and range) in a model at first. Then Katib will start multiple parallel Pods, which will take different value of those hyperparameters, to train and measure the model.
+
+For SQLFlow, each Katib pod will execute the following command:
+
+`repl -m "select * from c to train ... " | python sqlflow_submitter/couler/katib/xgboost_train.py ...`
```

Comment:
For my understanding, the Katib pod runs the SQL with the specified hyperparameter such as  `max_depth =3`, but the user input SQL is `max_depth =[2, 10]`, who implement this translation?